### PR TITLE
chore: improvements on Makefile

### PR DESCRIPTION
**Pull request by eordano**
- split different build systems for a faster build time
- better documentation
- rename *-parcels to *-scenes

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,162 +1,158 @@
...
+build-test-scenes: $(TEST_SCENES_GAMEJS_FILES) $(HELLMAP_GAMEJS_FILES) $(TEST_ECS_SCENE_GAMEJS_FILES) ## Build the test scenes
+
+# Entry points
+
+static/dist/%.js: build-essentials packages/entryPoints/%.ts
+	@$(COMPILER) $<
+	
+# Release
+
+DIST_ENTRYPOINTS := "static/dist/editor.js static/dist/unity.js static/dist/preview"
```
Comment (moliva) [inline Makefile]: Removing quotes as it makes `make` think that the filename is `"static/dist/editor.js` and won't find the target.
```suggestion
DIST_ENTRYPOINTS := static/dist/editor.js static/dist/unity.js static/dist/preview
```

File: Makefile [hunk 2]
```diff
@@ -1,162 +1,158 @@
...
+DIST_STATIC_FILES := "static/dist/preview.js static/export.html static/preview.html static/fonts static/images static/models static/unity
```
Comment (moliva) [inline Makefile]: Same as above.
```suggestion
DIST_STATIC_FILES := static/dist/preview.js static/export.html static/preview.html static/fonts static/images static/models static/unity 
```

File: Makefile [hunk 3]
```diff
@@ -1,162 +1,158 @@
...
+HELLMAP_SCENE_JSONS := $(wildcard public/hell-map/*/scene.json)
+SCENE_JSONS := $(PARCEL_SCENE_JSONS) $(ECS_SCENE_JSONS) $(HELLMAP_SCENE_JSONS)
 
-watch-single-no-server:
-	@echo '[{"name": "Debug","kind": "Webpack","file": "public/test-parcels/$(FILE)/game.ts","target": "web"}]' > build.single-debug.json
-	@node_modules/.bin/concurrently \
-		-n "entryPoints,debug-scene" \
-			"$(PARALLEL_COMPILER) build.entryPoints.json --watch" \
-			"$(PARALLEL_COMPILER) build.single-debug.json --watch"
+public/local-ipfs/mappings: ./scripts/createMockJson.js $(SCENE_JSONS)
```
Comment (moliva) [inline Makefile]: I'd argue to remove local-ipfs directory as the first step of this target, as we've had issues with preexistent contents for scenes that were later removed.
```suggestion
public/local-ipfs/mappings: ./scripts/createMockJson.js $(SCENE_JSONS)
	rm -r ./public/local-ipfs
```

**Review by moliva — APPROVED** [review]

**Comment by eordano** [comment]
@fmiras could you PTAL at this?

File: Makefile [hunk 4]
```diff
@@ -172,10 +169,47 @@ initialize-ecs-npm-link: build-sdk
 	ln -sf $(CWD)/static/unity-preview.html packages/decentraland-ecs/artifacts/unity-preview.html
 	cd packages/decentraland-ecs; npm link
 
+watch-cli: build-essentials ## Watch the files required for building the CLI
```
Comment (fmiras) [inline Makefile:186]: Awesome ❤️

**Review by fmiras — APPROVED** [review]
LGTM

**Comment by eordano** [comment]
@pravusjif can you verify `make watch` now works with the unity editor? works for me -- but it doesn't build the test scenes. You need to do `make test-scenes watch` the first time.

**Comment by pravusjif** [comment]
> @pravusjif can you verify `make watch` now works with the unity editor? works for me -- but it doesn't build the test scenes. You need to do `make test-scenes watch` the first time.

It's working great now!

**Comment by eordano** [comment]
@pbosio could you please check that this works correctly with the builder? You will want to use `make watch-builder`

**Comment by pbosio** [comment]
> @pbosio could you please check that this works correctly with the builder? You will want to use `make watch-editor`

seems to be working correctly with the builder 👍

**Pull request merged**